src/planet-a/SubdivTriangleMesh.h

```diff
@@ -0,0 +1,487 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2021 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#pragma once
+
+#include "SubdivSkeleton.h"
+
+#include <osp/types.h>
+
+#include <memory>
+#include <variant>
+
+namespace planeta
+{
+
+// ID for all chunks, from 0 to m_chunkMax
+enum class ChunkId : uint16_t {};
+
+// ID for all shared vertices; from 0 to m_sharedMax
+enum class SharedVrtxId : uint32_t {};
+
+// Indices of vertices, unaware of vertex size;
+// from 0 to (m_chunkMax*m_chunkVrtxCount + m_sharedMax)
+enum class VertexId : uint32_t {};
+
+// IDs for any chunk's shared vertices; from 0 to m_chunkVrtxSharedCount
+enum class ChunkLocalSharedId : uint16_t {};
+
+// IDs for any chunk's fill vertices; from 0 to m_chunkVrtxSharedCount
+enum class ChunkLocalFillId : uint16_t {};
+
+using SharedVrtxStorage_t = osp::IdRefCount<SharedVrtxId>::Storage_t;
+
+//-----------------------------------------------------------------------------
+
+/**
+ * Convert XY coordinates to a triangular number index
+ *
+ *  0
+ *  1  2
+ *  3  4  5
+ *  6  7  8  9
+ * 10 11 12 13 14
+ * ...
+ * x = right; y = down; (0, 0) = 0
+ *
+ * @param x [in]
+ * @param y [in]
+ * @return triangular number
+ */
+constexpr int xy_to_triangular(int x, int y) noexcept
+{
+    return y * (y + 1) / 2 + x;
```

Comment:
division by 2 has no percision loss, i assume?

Might help readers to put parens to clarify the order of operations.

Reply:
"y * (y + 1)" will always result in an even number. The order of operations here is already pretty basic, but I guess it doesn't hurt to add more parentheses.